src/libraries/System.Private.Xml/src/System/Xml/Xsl/Runtime/XslNumber.cs

```diff
@@ -192,7 +193,7 @@ public string FormatSequence(IList<XPathItem> val)
                     // Errata E24: It is an error if the number is NaN, infinite or less than 0.5; an XSLT processor may signal
                     // the error; if it does not signal the error, it must recover by converting the number to a string as if
                     // by a call to the 'string' function and inserting the resulting string into the result tree.
-                    return XPathConvert.DoubleToString(dblVal);
+                    return dblVal.ToString(CultureInfo.InvariantCulture);
```

Comment:
I'd suggest split this change into removing method by method. I.e. Remove just DoubleToString and add relevant test cases. And do so for all methods you're replacing here. If we regress here we'll need to revert entire changeset here as I doubt anyone will be willing to spend time investigating what went wrong otherwise.

Reply:
(you can remove all dead code immediately or in the first PR)

Reply:
Yeah I'm going to split this.